main/src/cgeo/geocaching/utils/MapMarkerUtils.java

```diff
@@ -697,4 +705,41 @@ public static void resetLists() {
         return result;
     }
 
+    public static Drawable getDTRatingMarker(final Resources res, final float difficulty, final float terrain) {
+        final int hashcode = new HashCodeBuilder().append(difficulty).append(terrain).toHashCode();
+
+        synchronized (overlaysCache) {
+            CacheMarker marker = overlaysCache.get(hashcode);
+            if (marker == null) {
+                marker = new CacheMarker(hashcode, createDTRatingMarker(res, difficulty, terrain));
+                overlaysCache.put(hashcode, marker);
+            }
+            return marker.getDrawable();
+        }
+    }
+
+    private static LayerDrawable createDTRatingMarker(final Resources res, final float difficulty, final float terrain) {
+        final Drawable background = DrawableCompat.wrap(ResourcesCompat.getDrawable(res, R.drawable.marker_rating_bg, null));
+        final InsetsBuilder insetsBuilder = new InsetsBuilder(res, background.getIntrinsicWidth(), background.getIntrinsicHeight());
+        insetsBuilder.withInset(new InsetBuilder(background));
+
+        final String packageName = CgeoApplication.getInstance().getPackageName();
+
+        Map<String, Integer> dt = new HashMap<>();
```

Comment:
Really necessary to create a map here with two entries? Might it be possible to extract the remainder of the method into a submethod and call it twice instead? This might increase readability